### PR TITLE
Changed filenames to tsx, added types

### PR DESCRIPTION
**Pull request by nadja-po**
Changed filenames to tsx, fixed all TypeScript errors, changed files according to TypeScript.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -6,9 +6,9 @@ import Footer from './components/molecules/Footer';
 import Gallery from './components/molecules/Gallery';
 import './App.scss';
 
-const App = () => {
-  const [dataJson, setDataJson] = useState([]);
-  const getData = () => {
+const App: React.FunctionComponent = () => {
+  const [dataJson, setDataJson] = useState<any>([]);
```
Comment (ArtemsBoroviks) [inline src/App.tsx]: Avoid to use "any" type

File: src/components/molecules/AboutPart/index.tsx [hunk 2]
```diff
@@ -3,9 +3,14 @@ import Title from '../../atoms/Title';
...
 import './style.scss';
 
-function AboutPart({
-  titleAbout, subtitleAbout, textAbout, imagesAbout,
-}) {
+type Props = {
+  titleAbout: string,
+  subtitleAbout: string,
+  textAbout: string,
+  imagesAbout: Array<{ src: string, alt: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/molecules/AboutPart/index.tsx]: Define interface for this, and use like this "Images[]"

File: src/components/molecules/Footer/index.tsx [hunk 3]
```diff
@@ -4,16 +4,21 @@ import Logo from '../../atoms/Logo';
 import Icon from '../../atoms/Icon';
 import './style.scss';
 
-function Footer({
-  textFooter, iconsFooter, logo, copyrightFooter,
-}) {
+type Props = {
+  textFooter: string,
+  iconsFooter: Array<{ id: number, src: string, alt: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/molecules/Footer/index.tsx]: as above define interface and use like this FooterIcons[]

File: src/components/molecules/Gallery/index.tsx [hunk 4]
```diff
@@ -2,7 +2,13 @@ import React from 'react';
 import Title from '../../atoms/Title';
 import './style.scss';
 
-function Gallery({ titleGallery, subtitleGallery, imagesGallery }) {
+type Props = {
+  titleGallery: string,
+  subtitleGallery: string,
+  imagesGallery: Array<{ id: number, src: string, alt: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/molecules/Gallery/index.tsx]: as above comment define interface

File: src/components/organisms/Cart/index.tsx [hunk 5]
```diff
@@ -3,21 +3,25 @@ import Icon from '../../atoms/Icon';
...
 
-const Cart = ({
-  updateCartState, updateMenuState, isMenuVisible, cartItems,
-}) => {
-  const [isCartVisible, setCartIsVisible] = useState(false);
+type Props = {
+  updateCartState: React.Dispatch<React.SetStateAction<boolean>>,
+  updateMenuState: React.Dispatch<React.SetStateAction<boolean>>,
+  isMenuVisible: boolean,
+  cartItems: Array<{ id: number, title: string, url: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Cart/index.tsx]: as above interface

File: src/components/organisms/Menu/index.tsx [hunk 6]
```diff
@@ -6,24 +6,31 @@ import burger from '../../../images/burger.svg';
 import close from '../../../images/close.svg';
 import './style.scss';
 
-const Menu = ({
+type Props = {
+  updateCartState: React.Dispatch<React.SetStateAction<boolean>>,
+  updateMenuState: React.Dispatch<React.SetStateAction<boolean>>,
+  isCartVisible: boolean,
+  menuItems: Array<{ id: number, title: string, url: string, button: boolean, onclick: boolean }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Menu/index.tsx]: interface

File: src/components/organisms/Menu/index.tsx [hunk 7]
```diff
@@ -6,24 +6,31 @@ import burger from '../../../images/burger.svg';
...
+  submenuItems: Array<{ id: number, title: string, url: string, button: boolean, onclick: boolean }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Menu/index.tsx]: interface

File: src/components/organisms/Navbar/index.tsx [hunk 8]
```diff
@@ -4,16 +4,21 @@ import Menu from '../Menu';
 import Logo from '../../atoms/Logo';
 import './style.scss';
 
-const Navbar = ({
-  logo, cartItems, menuItems, submenuItems,
-}) => {
+type Props = {
+  logo: string,
+  cartItems: Array<{ id: number, title: string, url: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Navbar/index.tsx]: interface

File: src/components/organisms/Navbar/index.tsx [hunk 9]
```diff
@@ -4,16 +4,21 @@ import Menu from '../Menu';
...
+  menuItems: Array<{ id: number, title: string, url: string, button: boolean, onclick: boolean }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Navbar/index.tsx]: interface

File: src/components/organisms/Navbar/index.tsx [hunk 10]
```diff
@@ -4,16 +4,21 @@ import Menu from '../Menu';
...
+  submenuItems: Array<{ id: number, title: string, url: string, button: boolean, onclick: boolean }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Navbar/index.tsx]: interface

File: src/components/organisms/Slider/index.tsx [hunk 11]
```diff
@@ -3,21 +3,27 @@ import Button from '../../atoms/Button';
 import CarouselText from '../../molecules/CarouselText';
 import './style.scss';
 
-const Slider = ({ titleCarousel, subtitleCarousel, imagesCarousel }) => {
-  const length = imagesCarousel && imagesCarousel.length;
+type Props = {
+  titleCarousel: string,
+  subtitleCarousel: string,
+  imagesCarousel: Array<{ id: number, src: string, alt: string }>,
```
Comment (ArtemsBoroviks) [inline src/components/organisms/Slider/index.tsx]: interface

**Review by ArtemsBoroviks — CHANGES_REQUESTED** [review]

File: src/components/molecules/AboutPart/index.tsx (hunk 2 above)
Comment (nadja-po) [reply src/components/molecules/AboutPart/index.tsx]: Added interface

File: src/components/molecules/Footer/index.tsx (hunk 3 above)
Comment (nadja-po) [reply src/components/molecules/Footer/index.tsx]: Added interface

File: src/components/molecules/Gallery/index.tsx (hunk 4 above)
Comment (nadja-po) [reply src/components/molecules/Gallery/index.tsx]: Added interface

File: src/components/organisms/Cart/index.tsx (hunk 5 above)
Comment (nadja-po) [reply src/components/organisms/Cart/index.tsx]: Added interface

File: src/components/organisms/Menu/index.tsx (hunk 6 above)
Comment (nadja-po) [reply src/components/organisms/Menu/index.tsx]: Added interface

File: src/components/organisms/Menu/index.tsx (hunk 7 above)
Comment (nadja-po) [reply src/components/organisms/Menu/index.tsx]: Added interface

File: src/components/organisms/Navbar/index.tsx (hunk 8 above)
Comment (nadja-po) [reply src/components/organisms/Navbar/index.tsx]: Added interface

File: src/components/organisms/Navbar/index.tsx (hunk 9 above)
Comment (nadja-po) [reply src/components/organisms/Navbar/index.tsx]: Added interface

File: src/components/organisms/Navbar/index.tsx (hunk 10 above)
Comment (nadja-po) [reply src/components/organisms/Navbar/index.tsx]: Added interface

File: src/components/organisms/Slider/index.tsx (hunk 11 above)
Comment (nadja-po) [reply src/components/organisms/Slider/index.tsx]: Added interface

File: src/App.tsx (hunk 1 above)
Comment (ArtemsBoroviks) [reply src/App.tsx]: Other thinks are fine, what about this?

Comment (nadja-po) [reply src/App.tsx]: I couldn't find the right type in this case.

Comment (ArtemsBoroviks) [reply src/App.tsx]: Check this topic
https://www.carlrippon.com/typed-usestate-with-typescript/

**Review by ArtemsBoroviks — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (nadja-po) [reply src/App.tsx]: Fixed: changed type Data instead any type, added checking for undefined.

**Pull request merged**